force-app/main/default/classes/TaskTriggerHandler.cls

```diff
@@ -0,0 +1,104 @@
+public without sharing class TaskTriggerHandler {
+  @testvisible
+  private static final Id recordTypeUS = Schema.SObjectType.Account.getRecordTypeInfosByDeveloperName()
+    .get('United_States')
+    .getRecordTypeId();
+
+  public static void handleAfterInsert(List<Task> newTasks) {
+    Set<Id> whatIds = filterTasks(newTasks);
+    if (whatIds.isEmpty()) {
+      return;
+    }
+    List<Account> accountsToUpdate = setTaskCountValues(
+      getRelatedAccountsWithOpenTasks(whatIds)
+    );
+    update accountsToUpdate;
```

Comment:
code duplication between the three handler methods. do you see a way to avoid that or do you opt for: its good as it is because....

Reply:
I could create a new method and put the common code inside it, like that:

private static void handleTasks(List<Task> tasks) {
    Set<Id> whatIds = filterTasks(tasks);
    if (whatIds.isEmpty()) {
      return;
    }
    List<Account> accountsToUpdate = setTaskCountValues(
      getRelatedAccountsWithOpenTasks(whatIds)
    );
    update accountsToUpdate;
  }


And then I could call this method from each handler method:
public static void handleAfterInsert(List<Task> newTasks) {
    handleTasks(newTasks);
  }
  
But what about the handleAfterUpdate method which has 2 params?
public static void handleAfterUpdate(
    List<Task> oldTasks,
    List<Task> newTasks
  ) {
    handleTasks(oldTasks);
    handleTasks(newTasks);
  }
  
But this doesn't seem like the best method since for each after update I'm calling each method twice, so I'm doing 2 queries, 2 updates etc.

Reply:
So how do you decide? :)

Reply:
In my opinion, handler methods are good as they are, because I can't use the new `handleTasks()` method only for `handleAfterInsert()` and `handleAfterDelete()` and leave the `handleAfterUpdate()` method as it is or call the `handleTasks()` on this method twice. I think these methods should follow the same line.

Reply:
Accepted :)